dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/schema/DubboBeanDefinitionParser.java

```diff
@@ -96,9 +96,15 @@ private static RootBeanDefinition parse(Element element, ParserContext parserCon
                 generatedBeanName = beanClass.getName();
             }
             id = generatedBeanName;
-            int counter = 2;
-            while (parserContext.getRegistry().containsBeanDefinition(id)) {
-                id = generatedBeanName + (counter++);
+            if (parserContext.getRegistry().containsBeanDefinition(id)) {
+                if (beanClass.equals(ServiceBean.class)) {
+                    throw new IllegalStateException("Duplicate service bean id " + id);
```

Comment:
I think it is only necessary to prohibit the automatic renaming of ServiceBean, and the problem of duplicate bean definition has been checked later.

```
if (!beanClass.equals(ServiceBean.class)) {
            int counter = 2;
            while (parserContext.getRegistry().containsBeanDefinition(id)) {
                id = generatedBeanName + (counter++);
            }
}
```

Reply:
Yeah，you are right

Reply:
i fixed it